internal/controller/logger_manager.go

```diff
@@ -0,0 +1,267 @@
+// Copyright 2024 Illumio, Inc. All Rights Reserved.
+
+package controller
+
+import (
+	"io"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+
+	pb "github.com/illumio/cloud-operator/api/illumio/cloud/k8scluster/v1"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/connectivity"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+const (
+	maxBufferSize = 100
+	flushInterval = 5 * time.Second
+)
+
+// BufferedGrpcWriteSyncer is a custom zap writesync that writes to a grpc stream
+// In case stream is not connected it will buffer to memory
+type BufferedGrpcWriteSyncer struct {
+	client    pb.KubernetesInfoService_SendLogsClient
+	conn      *grpc.ClientConn
+	buffer    []zapcore.Entry
+	mutex     sync.Mutex
+	done      chan struct{}
+	logger    *zap.SugaredLogger
+	logLevel  *zap.AtomicLevel
+	encoder   zapcore.Encoder
+	lostLogs  bool
+	lostBytes int
```

Comment:
You're counting log entries (which is good!), not bytes.
```suggestion
	lostLogEntries int
```